### PR TITLE
SW-5639 Log report validation failure reason

### PR DESCRIPTION
**Pull request by sgrimm**
When a user tried to submit a report and the report validation failed, we weren't
logging the failure reason which made the failure hard to diagnose after the fact.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#2146** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by tommylau523 — APPROVED** [review]

File: src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt [hunk 1]
```diff
@@ -298,32 +298,37 @@ class ReportStore(
...
-      saveSeedBankInfo(body)
-      saveNurseryInfo(body)
+        saveSeedBankInfo(body)
+        saveNurseryInfo(body)
 
-      eventPublisher.publishEvent(ReportSubmittedEvent(reportId, body))
+        eventPublisher.publishEvent(ReportSubmittedEvent(reportId, body))
+      }
+    } catch (e: Exception) {
+      log.info("Report $reportId cannot be submitted: ${e.message}")
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt:330]: Not part of this PR: Wondering if we should do an application-wide catch/log/throw.

Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt:330]: Exceptions that bubble up past the controller method are already logged (without stack traces, though, which maybe we should change). But this one was being caught by a try/catch in the controller method so the default logging didn't happen.

Auditing the controller methods and make sure any exceptions they catch get logged somewhere would probably be a good idea.
